src/providers/ipa/ipa_deskprofile_rules_util.c

```diff
@@ -0,0 +1,796 @@
+    if (ret != EOK) {
+        DEBUG(SSSDBG_CRIT_FAILURE,
+              "ipa_deskprofile_rule_check_memberuser() failed [%d]: %s\n",
+              ret, sss_strerror(ret));
+        goto done;
+    }
+
+    ret = ipa_deskprofile_rule_check_memberhost(tmp_ctx, rule, rule_name,
+                                                rule_prio, base_dn, hostname,
+                                                &host_prio, &hostgroup_prio);
+    if (ret != EOK) {
+        DEBUG(SSSDBG_CRIT_FAILURE,
+              "ipa_deskprofile_rule_check_memberhost() failed [%d]: %s\n",
+              ret, sss_strerror(ret));
+        goto done;
+    }
+
+    ret = ipa_deskprofile_get_normalized_rule_name(mem_ctx, rule_name,
+                                                   &normalized_rule_name);
+    if (ret != EOK) {
+        DEBUG(SSSDBG_CRIT_FAILURE,
+              "ipa_deskprofile_get_normalized_rule_name() failed [%d]: %s\n",
+              ret, sss_strerror(ret));
+        goto done;
+    }
+
+    ret = ipa_deskprofile_get_filename_path(tmp_ctx,
+                                            priority,
+                                            IPA_DESKPROFILE_RULES_USER_DIR,
+                                            domain,
+                                            username,
+                                            rule_prio,
+                                            user_prio,
+                                            group_prio,
+                                            host_prio,
+                                            hostgroup_prio,
+                                            normalized_rule_name,
+                                            extension,
+                                            &filename_path);
+    if (ret != EOK) {
+        DEBUG(SSSDBG_CRIT_FAILURE,
+              "ipa_deskprofile_get_filename_path() failed [%d]: %s\n",
+              ret, sss_strerror(ret));
+        goto done;
+    }
+
+    fp = fopen(filename_path, "w");
+    if (fp == NULL) {
+        ret = errno;
+        DEBUG(SSSDBG_TRACE_FUNC,
+              "Failed to create the Desktop Profile rule file \"%s\" "
+              "[%d]: %s\n",
+              filename_path, ret, sss_strerror(ret));
+        goto done;
+    }
+
+    fprintf(fp, "%s", data);
+
+    ret = fclose(fp);
+    if (ret != EOK) {
+        ret = errno;
+        DEBUG(SSSDBG_TRACE_FUNC,
+              "Failed to close the Desktop Profile rule file \"%s\" [%d]: %s\n",
+              filename_path, ret, sss_strerror(ret));
+        goto done;
+    }
+
+    ret = chown(filename_path, uid, gid);
+    if (ret != EOK) {
+        ret = errno;
+        DEBUG(SSSDBG_CRIT_FAILURE,
+              "Failed to own the Desktop Profile Rule file \"%s\" [%d]: %s\n",
+              filename_path, ret, sss_strerror(ret));
+        goto done;
+    }
+
+    ret = EOK;
+
+done:
+    talloc_free(tmp_ctx);
+    return ret;
+}
+
+errno_t
+ipa_deskprofile_rules_remove_user_dir(const char *user_dir)
```

Comment:
I didn't even know about the functions on src/tools/files.c

I got rid of the whole code that could be re-used from there in the new patchset (that's only locally for now).

So, fixed locally!